language.md

```diff
@@ -0,0 +1,543 @@
+This so-called meta-text can only be ended with a double quote and the same number of single quotes that it started with.
+
+Finally, you can use curly braces (`{}`) containing a value to insert a stringified version of the value into the string at that position.
+
+```candy
+"Hello!"
+"A somewhat
+  long
+  text."
+'"This is a meta text, where you can use " inside the text."'
+''"This is a double-meta text, allowing you to use "' inside it without ending it."''
+"Some {interpolation}."
+'"In meta texts, {{interpolation}} requires more curly braces, otherwise the values are {not interpolated}."'
+```
+
+### Symbols
+
+Symbols are uppercase identifiers that can only be compared for equality.
+
+```candy
+True
+Green
+Foo
+```
+
+### Structs
+
+Structs are mappings from keys to values (also known as dictionaries or hash maps in other languages).
+
+```candy
+[
+  Name: "Candy",
+  Foo: 42,
+]
+```
+
+TODO: Struct access using dot
+
+TODO: Modifying structs. Original idea: `{ Name: "Marcel", Age: 21 }` copied using `{ original | Name: "Jonas" }`
+
+### Closures
+
+Closures are pieces of code that can be executed.
+
+```candy
+identityFunction = { argument -> argument }
+longClosure = { foo ->
+  ...
+}
+```
+
+### More?
+
+TODO: Lists? Sets?
+
+- lists: [1, 2, 3]
+- sets: Clojure has %{ value }
+  - or like Toit? {hey, you, there} for set, empty map is {:}
+
+### Channel Ends
+
+Channels ends allow you to interact with a [channel](#concurrency).
+There are receive ends and send ends to receive and send data from a channel, respectively.
+
+## Variables
+
+Variables are named slots for storing values.
+You define a new variable using the equals sign `=`, like so:
+
+```candy
+foo = "Hello!"
+```
+
+This creates a new variable `foo` in the current scope and initializes it with the result of the expression following the `=`.
+Once a variable has been defined, it can never be reassigned.
+You can access variables by their name.
+
+```candy
+foo = "Hi!"
+bar = foo  # bar = "Hi!"
+```
+
+Variables only exist until the end of the scope they're defined in.
+
+```candy
+foo =
+  bar = hello  # error because hello doesn't exist yet
+  hello = 5
+  4
+bar = hello  # error because hello doesn't exist anymore
+```
+
+Variables at the top level of a file are visible to the module system.
+All other variables are local.
+
+Declaring a variable with the same name as another simply shadows that variable:
+From that point forward, the name will refer to the new variable.
+Note that this is different from reassigning to a variable.
+
+```candy
+foo = 5
+foo = 3
+# there is no way to get to the 5
+bar =
+  foo = 4
+# here, foo is still 3
+```
+
+## Functions
+
+Functions can either be defined using closure literals or by writing them as parameterized variables with arguments in front of the `=`.
+Both representations are equivalent with respect to what they do during runtime.
+
+```candy
+# Both these definitions are equivalent.
```

Comment:
Already stated above about their behavior during runtime. During compile-time, however, they differ with regards to fuzzing.